components/external_storage/src/local.rs

```diff
@@ -215,4 +214,22 @@ mod tests {
     fn test_url_of_backend() {
         assert_eq!(url_for(Path::new("/tmp/a")).to_string(), "local:///tmp/a");
     }
+
+    #[tokio::test]
+    async fn test_write_existed_file() {
+        let temp_dir = Builder::new().tempdir().unwrap();
+        let path = temp_dir.path();
+        let ls = LocalStorage::new(path).unwrap();
+
+        let filename = "existed.file";
+        let buf: &[u8] = b"pingcap";
+        let r = ls
+            .write(filename, UnpinReader(Box::new(buf)), buf.len() as _)
+            .await;
+        assert!(r.is_ok());
+        let r = ls
+            .write(filename, UnpinReader(Box::new(buf)), buf.len() as _)
```

Comment:
Will it full cover the former file, if we first write '1234', and next time write '432'?

Reply:
it's better check the content in this case

Reply:
> Will it full cover the former file, if we first write '1234', and next time write '432'?

If think the content could be `4324`

Reply:
> > Will it full cover the former file, if we first write '1234', and next time write '432'?
> 
> I had test it in above case. The content will be `432`. The new content will cover the old content, because the localStorage::write will `move` a temp file to target filename.

OK